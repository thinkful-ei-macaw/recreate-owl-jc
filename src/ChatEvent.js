import React from 'react'

const ChatEvent = (props) => {

  function getEventMessage() {
    switch(props.type) {
      case 'thumbs-up': return `${props.name} gave a thumbs up 👍`;
      case 'thumbs-down': return `${props.name} gave a thumbs down 👎`;
      case 'raise-hand' : return `${props.name} raised their hand ✋`;
      case 'clap' : return `${props.name} clapped 👏`;
      case 'join' : return `${props.name} joined`;
      case 'leave' : return `${props.name} left`;
      case 'join-stage' : return `${props.name} joined the stage`;
      case 'leave-stage' : return `${props.name} left the stage`;
      default : return '';
    }
  }

  function getMessage() {
    if(props.message) {
      return <div>{props.name} -- {props.message} -- {new Date(props.time).toLocaleTimeString()}</div>
    } else {
      return getEventMessage();
    }
  }

  const message = getMessage();

return (
<div>
  {message}
</div>
);
}

export default ChatEvent;