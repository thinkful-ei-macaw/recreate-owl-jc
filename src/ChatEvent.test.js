import React from 'react';
import ReactDOM from 'react-dom';
import ChatEvent from './ChatEvent';
import renderer from 'react-test-renderer';

describe('Chat event Component tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatEvent />, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders message chats appropriately', () => {
      const tree = renderer
      .create(<ChatEvent 
        key={3}
        name='Bob'
        type='message'
        message='Hello world'
        time={1548852646559}
        timestamp={1548852484247}
        />)
      .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders event messages appropriately', () => {
      const tree = renderer
      .create(<ChatEvent 
        key={5}
        name='Cindy'
        type='raise-hand'
        timestamp={1548852544247}
        />)
      .toJSON();
      expect(tree).toMatchSnapshot();
    });

})
