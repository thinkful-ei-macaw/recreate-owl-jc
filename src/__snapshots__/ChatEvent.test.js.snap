// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat event Component tests renders event messages appropriately 1`] = `
<div>
  Cindy raised their hand âœ‹
</div>
`;

exports[`Chat event Component tests renders message chats appropriately 1`] = `
<div>
  <div>
    Bob
     -- 
    Hello world
     -- 
    7:50:46 AM
  </div>
</div>
`;
